pkg/comp-functions/functions/vshn-postgres-func/loadBalancer.go

```diff
@@ -0,0 +1,100 @@
+package vshnpostgres
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/crossplane/crossplane/apis/apiextensions/fn/io/v1alpha1"
+	vshnv1 "github.com/vshn/appcat/apis/vshn/v1"
+	"github.com/vshn/appcat/pkg/comp-functions/runtime"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+// AddLoadBalancerIPToConnectionDetails changes the desired state of a FunctionIO
+func AddLoadBalancerIPToConnectionDetails(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+	log := controllerruntime.LoggerFrom(ctx)
+
+	comp := &vshnv1.VSHNPostgreSQL{}
+	err := iof.Desired.GetComposite(ctx, comp)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Cannot get composite from function io", err)
+	}
+	// Wait for the next reconciliation in case instance namespace is missing
+	if comp.Status.InstanceNamespace == "" {
+		return runtime.NewWarning(ctx, "Composite is missing instance namespace, skipping transformation")
+	}
+
+	// if there is nothin to do, return early
+	if comp.Spec.Parameters.Network.ServiceType != "LoadBalancer" {
+		return runtime.NewNormal()
+	}
+	log.Info("Getting connection secret from managed kubernetes object")
+	s := &v1.Secret{}
+
+	err = iof.Observed.GetFromObject(ctx, s, connectionSecretResourceName)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Cannot get connection secret object", err)
+	}
+
+	log.Info("Setting ExternalLoadBalancerIP variable into connection secret")
+	primary, replicas, err := getLoadBalancerIPs(comp.Status.InstanceNamespace)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Cannot get load balancer IPs", err)
+	}
+	iof.Desired.PutCompositeConnectionDetail(ctx, v1alpha1.ExplicitConnectionDetail{
+		Name:  "PRIMARY_IP",
+		Value: primary,
+	})
+	iof.Desired.PutCompositeConnectionDetail(ctx, v1alpha1.ExplicitConnectionDetail{
+		Name:  "REPLICAS_IP",
+		Value: replicas,
+	})
+
+	return runtime.NewNormal()
+}
+
+func getLoadBalancerIPs(instanceNamespace string) (primary string, replicas string, err error) {
+	// Build the Kubernetes config
+	// stackgres just creates services in the same namespace as the instance
+	// therefore I need to manually check IP addresses and then return them
+	config, err := rest.InClusterConfig()
```

Comment:
I'm a bit torn with this. On one hand it's easier to get the values you want with this approach, on the other hand it breaks crossplane's concepts how to interact with the kubernetes objects.

We can shoot ourselves in the foot really badly if we start to rely on the kubeclient directly. Crossplane themselves advise us not to do that:

> Don’t bypass providers. Composition Functions are designed to tell Crossplane how to orchestrate managed resources - not to directly orchestrate external systems.

https://docs.crossplane.io/master/concepts/composition-functions/

So even though it's a bit more complicated I would do it via `provider-kubernetes` to get these values here.